db/users.go

```diff
@@ -20,52 +20,33 @@ import (
 	ch "github.com/couchbase/sync_gateway/channels"
 )
 
-// Struct that configures settings of a User/Role, for UpdatePrincipal.
-// Also used in the rest package as a JSON object that defines a User/Role within a DbConfig
-// and structures the request/response body in the admin REST API for /db/_user/*
+// PrincipalConfig represents a user/role as a JSON object.
+// Used to define a user/role within DbConfig, and structures the request/response body in the admin REST API
+// for /db/_user/*
```

Comment:
There seems to be some overlap between PrincipalConfig and PrincipalUpdates.  PrincipalUpdates doesn't include the read-only fields used for REST API responses, but aside from that there's a lot of commonality.  Did you consider more of a straight migration of PrincipalConfig from db to auth?  If so, what were the obstacles to that approach?

Reply:
It's an interesting thought, and I've pushed dcf8a24d101e88b9b13736893e7bcc7b0d0af8d3 which makes that change (and we can roll back if we decide it's not desirable). The downsides that I can see are:
* the read-only fields, as you called out (I had to update a couple tests that were incorrectly using `Channels` instead of `ExplicitChannels`)
* its structure becomes part of the admin API so we have less freedom to change it - having two separate structs maintains a separation between the public API and our implementation details
* it's a fairly noisy diff (though hopefully a one-off)

I don't feel strongly either way.